redaxo/src/core/lib/select.php

```diff
@@ -226,18 +226,18 @@ public function countOptions()
     /**
      * Fügt Optionen anhand der Übergeben SQL-Select-Abfrage hinzu.
      */
-    public function addSqlOptions($query)
+    public function addSqlOptions($query, $db=1)
     {
-        $sql = rex_sql::factory();
+        $sql = rex_sql::factory($db);
         $this->addOptions($sql->getArray($query, [], PDO::FETCH_NUM));
     }
 
     /**
      * Fügt Optionen anhand der Übergeben DBSQL-Select-Abfrage hinzu.
      */
-    public function addDBSqlOptions($query)
+    public function addDBSqlOptions($query, $db=1)
```

Comment:
Ich würde es bei dieser Funktion eher nicht hinzufügen.
Denn dies ist ja die von Markus erwähnte Variante, die die DB-Angabe innerhalb der Query anbietet.
Und da die andere Methode ja dann schon die explizite Angabe über zweiten Parameter unterstützt, brauchen wir es ja eigentlich hier nicht, oder?
(Hier gäbe es sonst eventuell zwei unterschiedliche DB-IDs in Query und zweitem Parameter)

Reply:
Ist logisch, hab´s rausgenommen. Die Änderung ist also nur noch für *addSqlOptions* im PR.